src/liquid/components/ld-select/ld-select.tsx

```diff
@@ -630,63 +703,69 @@ export class LdSelect implements InnerFocusable {
   private handleEnd = (ev) => {
     // Move focus to the last option.
     ev.preventDefault()
-    const options = Array.from(
+    const visibleOptions = Array.from(
       this.listboxRef.querySelectorAll('ld-option-internal')
-    )
-    if (document.activeElement !== options[options.length - 1]) {
-      options[options.length - 1].focusOption()
+    ).filter((option) => !this.optHidden(option))
+    if (document.activeElement !== visibleOptions[visibleOptions.length - 1]) {
+      visibleOptions[visibleOptions.length - 1].focusInner()
     }
   }
 
   private selectAndFocus = (
     ev: KeyboardEvent,
-    ldOption: HTMLLdOptionInternalElement
+    opt: HTMLLdOptionInternalElement | HTMLLdOptgroupInternalElement
   ) => {
-    if (!ldOption) return
+    if (!opt) return
 
     if (this.multiple && ev.shiftKey) {
       if (
-        this.isLdOption(document.activeElement) &&
+        this.isLdOptionInternal(document.activeElement) &&
+        !this.isLdOptgroupInternal(document.activeElement) &&
         !document.activeElement.hasAttribute('selected')
       ) {
         document.activeElement.dispatchEvent(
           new KeyboardEvent('keydown', { key: ' ' })
         )
       }
-      ldOption.focusOption()
-      if (!ldOption.hasAttribute('selected')) {
-        ldOption.dispatchEvent(new KeyboardEvent('keydown', { key: ' ' }))
+      opt.focusInner()
+      if (!opt.hasAttribute('selected') && !this.isLdOptgroupInternal(opt)) {
+        opt.dispatchEvent(new KeyboardEvent('keydown', { key: ' ' }))
       }
     } else {
-      ldOption.focusOption()
+      opt.focusInner()
```

Comment:
Can we move the 2 `opt.focusInner()` calls out of the if-else-clause, or is order important here? This would potentially spare us the `else`.